packages/actor-http-fetch/lib/ActorHttpFetch.ts

```diff
@@ -76,12 +76,46 @@ export class ActorHttpFetch extends ActorHttp {
 
     try {
       requestInit = await this.fetchInitPreprocessor.handle(requestInit);
+      let retryCount: number = action.context?.get(KeysHttp.httpRetryCount) ?? 0;
 
-      // Perform request
       const customFetch: ((input: RequestInfo, init?: RequestInit) => Promise<Response>) | undefined = action
         .context?.get(KeysHttp.fetch);
 
-      const response = await (customFetch || fetch)(action.input, requestInit);
+      // Actual fetch respecting timeout and retry count.
+      const getResponse = async(): Promise<Response> => {
+        // When retry count is greater than 0, record the errors for failed requests.
+        const errors: unknown[] = [];
+        while (retryCount-- >= 0) {
+          try {
+            return await (customFetch || fetch)(action.input, requestInit);
+          } catch (error: unknown) {
+            // If the fetch was aborted by timeout, we won't retry.
+            if (requestInit.signal?.aborted) {
+              throw error;
+            }
+            errors.push(error);
+            // Wait for specified delay.
+            if (retryCount >= 0) {
+              await new Promise((resolve, reject) => {
+                // Cancel waiting, if timeout is reached.
+                requestInit.signal?.addEventListener('abort', () => {
+                  reject(new Error('Fetch aborted by timeout.'));
```

Comment:
```suggestion
                requestInit.signal?.on('abort', reason => {
                  reject(reason ?? new Error('Request aborted before timeout.'));
```

Reply:
Regarding the `numberOfRetries` variable of the first comment. That's for passing the request on the third attempt. Or is there a jest-ier way of keeping track of this?

Regarding the second comment with the `signal?.on(...))` proposal - that does not seem to work as `signal` only exposes a lambda called `onabort` which is however undefined in that context.

Reply:
> Regarding the numberOfRetries variable of the first commen

Oh - that makes sense! Apologies for the slightly rushed review on my end last night :)

> Regarding the second comment with the signal?.on(...)) proposal

Ok - no worries; I'd assumed this was extending the `EventEmitter` object but I guess not.